### PR TITLE
[stable/jenkins] Add existingSecret to Jenkins backup AWS credentials

### PR DESCRIPTION
**Pull request by hajowieland**
#### What this PR does / why we need it:

Add option to allow specifying an existing Kubernetes secret for AWS credentials used by the Jenkins backup container.

#### Checklist

- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md

### CODE REVIEW

**Comment by maorfr** [comment]
hey @hajowieland,

i noticed that you added yourself to the maintainers section.
have you spoke to one of the existing maintainers about it?

File: stable/jenkins/values.yaml [hunk 1]
```diff
@@ -440,6 +440,14 @@ backup:
   # Additional arguments for kube-tasks
   # Ref: https://github.com/nuvo/kube-tasks#simple-backup
   extraArgs: []
+  # Add existingSecret for AWS credentials
+  existingSecret: {}
+  ## Example for using an existing secret
+   # jenkinsaws:
+  ## Use this key for AWS access key ID
+     # awsaccesskey: jenkins_aws_access_key
```
Comment (torstenwalter) [inline stable/jenkins/values.yaml:479]: Would be great if we can permanently remove the comment and change the structure to something like this:

```
backup:
  aws:
    # Use an existing secret which contains the ACCESS KEY and SECRET ACCESS KEY
    secretName:
    # Use this key for AWS access key ID
    awsaccesskey: aws_access_key
    # Use this key for AWS secret access key
    awssecretkey: aws_secret_key
```

That way you can still check 

{{- if .Values.backup.aws.secretName }}

**Review by torstenwalter — CHANGES_REQUESTED** [review]

File: stable/jenkins/values.yaml (hunk 1 above)
Comment (hajowieland) [reply stable/jenkins/values.yaml]: Hy @torstenwalter ,
no, this was by mistake, I thought I have to add myself to the maintainer list if I make any changes to the chart.
Please remove the line.

Comment (hajowieland) [reply stable/jenkins/values.yaml:479]: But then we can't specify the secret name (or have to hardcode it), right?
My thought was to be as flexible as possible, so a user can specify an existing secret with AWS credentials and the respective keys.

**Comment by hajowieland** [comment]
> hey @hajowieland,
> 
> i noticed that you added yourself to the maintainers section.
> have you spoke to one of the existing maintainers about it?

Hy @maorfr 
no, this was by mistake, I thought I have to add myself to the maintainer list if I make any changes to the chart.
Please remove the line.

File: stable/jenkins/values.yaml (hunk 1 above)
Comment (torstenwalter) [reply stable/jenkins/values.yaml:479]: @hajowieland I think we misunderstood each other. I did not means to enter the values for the secrets directly within the values.yaml, but just the keys under which the secrets could be looked up. Similar to what you did, but with the difference that a consumer just needs to specifiy the name of the secret . 

Maybe you have a better name to make it clear that those are just the keys. Of course we could also offer the option to real secrets and generate the needed kubernetes secret in that case.
For this we need good names within the values.yaml to ensure people understand which key is the actual secret and which one is just a key within the kubernetes secret used for lookup.

**Review by torstenwalter — COMMENTED** [review]
Would you mind changing the code so that one can still configure via which keys the real aws secrets are stored within the Kubernetes secret, but having default values already defined within values.yaml?

**Comment by maorfr** [comment]
/ok-to-test
/lgtm

thanks for contributing!
this is actually a really good idea!

**Pull request merged**